lib/account/account.controllers.js

```diff
@@ -17,29 +17,53 @@ module.exports = (models) => {
 
       return account;
     },
-    // getAccounts: async function(request, h) {
-    //   return models.Account.findAll()
-    //   .then(response => {
-    //     return response;
-    //   })
-    //   .catch(error => {
-    //     return error;
-    //   });
-    // },
+
+    getAccounts: async function(request, h) {
+      let returnAccounts;
+      try {
+        returnAccounts = await getAccounts();
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccounts(request.query). ' + e);
+      }
+      return h
+        .response(returnAccounts);
+    },
+
+    getAccount: async function(request, h) {
+      let returnAccount;
+      try {
+        returnAccount = await getAccount(request.params.id);
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccount(request.query). ' + e);
+      }
+      return h
+        .response(returnAccount);
+    },
+
     postAccount: async function(request, h) {
       let credentials = request.auth.credentials;
-
       let account;
+
+      // Check to see that a user has the proper account credentials
       try {
-        account = await getAccount(credentials);
+        account = await getCurrentAccount(credentials);
       } catch (e) {
         throw Boom.badImplementation('Error during postAccount - getAccount(credentials). ' + e);
       }
-
+      // Check to see that an account doesn't already exist for the retrieved credentials
       if (account !== null) {
         throw Boom.conflict("Account already exists for provided credentials");
       }
 
+      // // Check to see if an account with the given username exists
+      // let accountUsername = await lookupAccountFromUsername(request.payload.username);
+      // if (Array.isArray(accountUsername) && accountUsername.length) {
+      //   let error = Boom.badData('Account with that username already exists');
+      //   error.output.headers["Content-Location"] = accountUsername[0].id;
+      //   throw error;
+      // }
```

Comment:
We talked it through in slack that the username/displayname should not be required on post, but I'm still debating on whether it should still be _possible_ to submit it (vs having to submit it in a second patch request). It sounds like tricia is coding the front end to just assume the patch request and not submit anything on post, but just wanted to let you know where my head is at. No *need* to change this!